microcosm_postgres/encryption/v2/column.py

```diff
@@ -71,19 +124,32 @@ def _prop(self):
             return encoder.decode(encryptor.decrypt(encrypted))
 
         def _prop_setter(self, value) -> None:
-            encrypted = encryptor.encrypt(encoder.encode(value))
+            encoded = encoder.encode(value)
+            encrypted = encryptor.encrypt(encoded)
             if encrypted is None:
                 setattr(self, unencrypted_field, value)
                 setattr(self, encrypted_field, None)
+                if hasattr(self, beacon_field):
+                    setattr(self, beacon_field, None)
                 return
 
             setattr(self, encrypted_field, encrypted)
             setattr(self, unencrypted_field, None)
+            if hasattr(self, beacon_field):
+                setattr(self, beacon_field, encryptor.beacon(encoded))
+
+        def _prop_comparator(cls) -> BeaconComparator | None:
```

Comment:
What happens when we return None here as the comparator?

Reply:
I've updated this so we don't return None. Type was incorrect.